src/libs/actions/IOU.js

```diff
@@ -949,12 +949,64 @@ function editMoneyRequest(transactionID, transactionThreadReportID, transactionC
     const iouReport = allReports[`${ONYXKEYS.COLLECTION.REPORT}${transactionThread.parentReportID}`];
+    const chatReport = allReports[`${ONYXKEYS.COLLECTION.REPORT}${iouReport.chatReportID}`];
     const isFromExpenseReport = ReportUtils.isExpenseReport(iouReport);
 
     // STEP 2: Build new modified expense report action.
     const updatedReportAction = ReportUtils.buildOptimisticModifiedExpenseReportAction(transactionThread, transaction, transactionChanges, isFromExpenseReport);
     const updatedTransaction = TransactionUtils.getUpdatedTransaction(transaction, transactionChanges, isFromExpenseReport);
+
     // STEP 3: Compute the IOU total and update the report preview message so LHN amount owed is correct
+    // Should only update if the transaction matches the currency of the report, else we wait for the update
+    // from the server with the currency conversion
+    let updatedIouReport = null;
+    const updatedChatReport = {...chatReport};
+    if (updatedTransaction.currency === iouReport.currency && updatedTransaction.modifiedAmount) {
+        if (ReportUtils.isExpenseReport(iouReport)) {
+            updatedIouReport = {...iouReport};
+            updatedIouReport.total -= TransactionUtils.getAmount(transaction, true);
+            updatedIouReport.total += TransactionUtils.getAmount(updatedTransaction, true);
+        } else {
+            updatedIouReport = IOUUtils.updateIOUOwnerAndTotal(
+                iouReport,
+                updatedReportAction.actorAccountID,
+                TransactionUtils.getAmount(transaction, false),
+                TransactionUtils.getCurrency(transaction),
+                true,
+            );
+            updatedIouReport = IOUUtils.updateIOUOwnerAndTotal(
+                updatedIouReport,
+                updatedReportAction.actorAccountID,
+                TransactionUtils.getAmount(updatedTransaction, false),
+                TransactionUtils.getCurrency(updatedTransaction),
+                false,
+            );
+        }
+
+        updatedIouReport.cachedTotal = CurrencyUtils.convertToDisplayString(updatedIouReport.total, updatedTransaction.currency);
+
+        // Update the last message of the IOU report
+        const lastMessage = ReportUtils.getIOUReportActionMessage(
+            iouReport.reportID,
+            CONST.IOU.REPORT_ACTION_TYPE.CREATE,
+            updatedIouReport.total,
+            '',
+            updatedTransaction.currency,
+            '',
+            false,
+        );
+        updatedIouReport.lastMessageText = lastMessage[0].text;
+        updatedIouReport.lastMessageHtml = lastMessage[0].html;
+
+        // Update the last message of the IOU chat report
+        const messageText = Localize.translateLocal('iou.payerOwesAmount', {
+            payer: updatedIouReport.managerEmail,
+            amount: CurrencyUtils.convertToDisplayString(updatedIouReport.total, updatedIouReport.currency),
```

Comment:
Ithe IOU should be capitalized
```suggestion
            amount: CurrencyUtils.convertToDisplayString(updatedIOUReport.total, updatedIOUReport.currency),
```